File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        mn = min(strs)
        k = len(mn)
        
        res = ""
        
        for i,c in enumerate(mn):
            valid = True
            
            for word in strs:
                if word != mn and word[i] != c:
                    valid = False
                    
            if valid:
                res+=c
                
            else:
                break
                
        return res

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res = "";
        
        string mn = *min_element(begin(strs),end(strs));
        
        for(int i=0;i<mn.size();i++){
            bool valid = true;
            for(auto &x: strs){
                if(x!=mn and x[i] != mn[i])
                    valid = false;
            }
            if(valid)
                res+=mn[i];
            else
                break;
        }
        
        return res;
    }
};